Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalMixRotatedLayer.cc

```diff
@@ -0,0 +1,597 @@
+                                      << cassette << " Shift " << cshift.first << ":" << cshift.second;
+#endif
+        std::string name = namesTop_[ii] + "L" + std::to_string(copy) + "F" + std::to_string(k);
+        ++k;
+        dd4hep::Solid solid = dd4hep::Tube(r1, r2, hthickl, phi1, phi2);
+        ns.addSolidNS(ns.prepend(name), solid);
+        dd4hep::Volume glog1 = dd4hep::Volume(solid.name(), solid, matter1);
+        ns.addVolumeNS(glog1);
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalMixRotatedLayer: " << glog1.name() << " Tubs made of "
+                                      << materialTop_[ii] << " of dimensions " << cms::convert2mm(r1) << ", "
+                                      << cms::convert2mm(r2) << ", " << cms::convert2mm(hthickl) << ", "
+                                      << convertRadToDeg(phi1) << ", " << convertRadToDeg(phi2);
+#endif
+        dd4hep::Position tran(cshift.first, cshift.second, zpos);
+        glog.placeVolume(glog1, copy, tran);
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalMixRotatedLayer: Position " << glog1.name() << " number " << copy
+                                      << " in " << glog.name() << " at (0, 0, " << cms::convert2mm(zpos)
+                                      << ") with no rotation";
+#endif
+      }
+      ++copyNumberTop_[ii];
+      zpos += hthickl;
+    }
+    if (std::abs(thickTot - thick) > tol2_) {
+      if (thickTot > thick) {
+        edm::LogError("HGCalGeom") << "Thickness of the partition " << cms::convert2mm(thick) << " is smaller than "
+                                   << cms::convert2mm(thickTot)
+                                   << ": thickness of all its components in the top part **** ERROR ****";
+      } else {
+        edm::LogWarning("HGCalGeom") << "Thickness of the partition " << cms::convert2mm(thick)
+                                     << " does not match with " << cms::convert2mm(thickTot)
+                                     << " of the components in top part";
+      }
+    }
+
+    // Make the bottom part next
+    int layer = (copyM - firstLayer_);
+    static const double sqrt3 = std::sqrt(3.0);
+    int layercenter = (layerTypes_[layer] == HGCalTypes::CornerCenteredLambda)
+                          ? HGCalTypes::CornerCenterYp
+                          : ((layerTypes_[layer] == HGCalTypes::CornerCenteredY) ? HGCalTypes::CornerCenterYm
+                                                                                 : HGCalTypes::WaferCenter);
+    int layertype = (layerTypes_[layer] == HGCalTypes::WaferCenteredBack) ? 1 : 0;
+    int firstWafer = waferLayerStart_[layer];
+    int lastWafer = ((layer + 1 < static_cast<int>(waferLayerStart_.size())) ? waferLayerStart_[layer + 1]
+                                                                             : static_cast<int>(waferIndex_.size()));
+    double r = 0.5 * (waferSize_ + waferSepar_);
+    double R = 2.0 * r / sqrt3;
```

Comment:
`r` and `R` are not clear variable names. It would be better to call them `rMin` and `rMax` or something like that.
It's not essential to change the names.